frontend/src/components/promote-button.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {
+    Button,
+    Heading,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+    SimpleGrid,
+    useDisclosure,
+    Text,
+    useToast,
+} from '@chakra-ui/react';
+import useAuth from '@hooks/use-auth';
+import type { Registration } from '@api/registration';
+
+interface Props {
+    registration: Registration;
+}
+
+const PromoteButton = ({ registration }: Props) => {
+    const { idToken, signedIn } = useAuth();
+
+    const { isOpen: isOpenPromote, onOpen: onOpenPromote, onClose: onClosePromote } = useDisclosure();
+
+    const toast = useToast();
+
+    const BACKEND_URL = process.env.NEXT_PUBLIC_BACKEND_URL ?? 'http://localhost:8080';
+
+    const handlePromote = async () => {
+        if (!signedIn || !idToken) {
+            toast({
+                title: 'Du er ikke logget inn.',
+                status: 'error',
+                isClosable: true,
+            });
+            return;
+        }
+
+        const responsePromote = await fetch(
+            `${BACKEND_URL}/registration/promote/noemail/${registration.slug}/${registration.email}`,
+            {
+                method: 'POST',
+                headers: {
+                    Authorization: `Bearer ${idToken}`,
+                },
+            },
+        );
+
+        if (responsePromote.ok) {
+            onClosePromote();
+            toast({
+                title: 'Bruker ble promotert.',
+                status: 'success',
+                isClosable: true,
+            });
+        } else {
+            toast({
+                title: 'Kunne ikke promotere.',
+                status: 'error',
+                isClosable: true,
+            });
+        }
+    };
+
+    const handleEmail = async () => {
+        if (!signedIn || !idToken) {
+            toast({
+                title: 'Du er ikke logget inn.',
+                status: 'error',
+                isClosable: true,
+            });
+            return;
+        }
+
+        const responseEmail = await fetch(
+            `${BACKEND_URL}/registration/promote/email/${registration.slug}/${registration.email}`,
+            {
+                method: 'POST',
+                headers: {
+                    Authorization: `Bearer ${idToken}`,
+                },
+            },
+        );
+
+        if (responseEmail.ok) {
+            onClosePromote();
+            toast({
+                title: 'Email er sendt.',
+                status: 'success',
+                isClosable: true,
+            });
+        } else {
+            toast({
+                title: 'Kunne ikke sende email.',
+                status: 'error',
+                isClosable: true,
+            });
+        }
+    };
+
+    return (
+        <div>
```

Comment:
```suggestion
        <>
```

Kan gjøres om til en React Fragment